playbooks/reboot_cluster.yml

```diff
@@ -46,3 +54,58 @@
     service:
       name: mysql
       state: started
+
+- hosts: compute:!controller
+  serial: 1
+  tasks:
+  - name: Reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    ignore_errors: true
+
+  - name: Wait for SSH to come back
+    wait_for: host={{ ansible_default_ipv4.address }}
+              port={{ ansible_ssh_port|default(22) }}
+              delay=15
+
+- hosts: swift
+  serial: 1
+  tasks:
+  - name: Reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    ignore_errors: true
+
+  - name: Wait for SSH to come back
+    wait_for: host={{ ansible_default_ipv4.address }}
+              port={{ ansible_ssh_port|default(22) }}
+              delay=15
+
+  - name: swift health check
+    command: sudo /etc/sensu/plugins/check-swift-dispersion.py
+    register: result
+    until: result | succeeded
+    retries: 10
+    
+
+- hosts: ceph
```

Comment:
There are two groups to be concerned with. `ceph_monitors` and `ceph_osds`. Each perform different things and will have different health checks to use to validate them. I'll give you some pointers on Slack.